src/components/cc-invoice/cc-invoice.types.d.ts

```diff
@@ -0,0 +1,18 @@
+import {Invoice} from "../common.types";
+
+export type InvoiceState = InvoiceStateLoading | InvoiceStateError | InvoiceStateLoaded;
+
+interface InvoiceStateLoading {
+  state: 'loading';
+  number?: string;
+}
+
+interface InvoiceStateError {
+  state: 'error';
+  number: string;
+}
+
+interface InvoiceStateLoaded {
+  state: 'loaded';
+  invoice: Invoice;
```

Comment:
suggestion: in the following situations, we avoided the `invoice.invoice` by naming the property `value` but it was for lists. I think we should rename `invoice` to `value`. WDYT?

* src/components/cc-pricing-header/cc-pricing-header.types.d.ts
  * `PricingZoneStateLoaded` with `value: Zone[]`
* src/components/cc-email-list/cc-email-list.types.d.ts
  * `EmailListStateLoaded` with `value: EmailList`
* src/components/cc-tcp-redirection-form/cc-tcp-redirection-form.types.d.ts
  * `TcpRedirectionFormStateLoaded` with `value: TcpRedirectionState[]`